controllers/component_build_controller_pac.go

```diff
@@ -482,370 +479,166 @@ func generateMergeRequestSourceBranch(component *appstudiov1alpha1.Component) st
 
 // ConfigureRepositoryForPaC creates a merge request with initial Pipelines as Code configuration
 // and configures a webhook to notify in-cluster PaC unless application (on the repository side) is used.
-func (r *ComponentBuildReconciler) ConfigureRepositoryForPaC(ctx context.Context, component *appstudiov1alpha1.Component, config map[string][]byte, webhookTargetUrl, webhookSecret string) (prUrl string, err error) {
+func (r *ComponentBuildReconciler) ConfigureRepositoryForPaC(ctx context.Context, component *appstudiov1alpha1.Component, pacConfig map[string][]byte, webhookTargetUrl, webhookSecret string) (prUrl string, err error) {
 	log := ctrllog.FromContext(ctx).WithValues("repository", component.Spec.Source.GitSource.URL)
 	ctx = ctrllog.IntoContext(ctx, log)
 
 	gitProvider, _ := gitops.GetGitProvider(*component)
-	isAppUsed := gitops.IsPaCApplicationConfigured(gitProvider, config)
-
-	var accessToken string
-	if !isAppUsed {
-		accessToken = strings.TrimSpace(string(config[gitops.GetProviderTokenKey(gitProvider)]))
+	repoUrl := component.Spec.Source.GitSource.URL
+
+	gitClient, err := gitproviderfactory.CreateGitClient(gitproviderfactory.GitClientConfig{
+		PacSecretData:             pacConfig,
+		GitProvider:               gitProvider,
+		RepoUrl:                   repoUrl,
+		IsAppInstallationExpected: true,
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to create git client for %s provider", gitProvider)
 	}
 
-	// https://github.com/owner/repository
-	gitSourceUrlParts := strings.Split(strings.TrimSuffix(component.Spec.Source.GitSource.URL, ".git"), "/")
-
-	commitMessage := "Appstudio update " + component.Name
-	branch := generateMergeRequestSourceBranch(component)
-	mrTitle := "Appstudio update " + component.Name
-	mrText := mergeRequestDescription
-	authorName := "redhat-appstudio"
-	authorEmail := "rhtap@redhat.com"
-
-	var baseBranch string
-	if component.Spec.Source.GitSource != nil {
-		baseBranch = component.Spec.Source.GitSource.Revision
+	baseBranch := component.Spec.Source.GitSource.Revision
+	if baseBranch == "" {
+		baseBranch, err = gitClient.GetDefaultBranch(repoUrl)
+		if err != nil {
+			return "", nil
```

Comment:
why return `nil` when the error was detected?

Reply:
It's a mistake, thank you for letting me know.